src/main/host/syscall/futex.c

```diff
@@ -159,32 +160,45 @@ SysCallReturn syscallhandler_futex(SysCallHandler* sys, const SysCallArgs* args)
     switch (operation) {
         case FUTEX_WAIT: {
             trace("Handling FUTEX_WAIT operation %i", operation);
-            return _syscallhandler_futexWaitHelper(sys, uaddrptr, val, timeoutptr);
+            return _syscallhandler_futexWaitHelper(
+                sys, uaddrptr, val, timeoutptr, TIMEOUT_RELATIVE);
         }
 
         case FUTEX_WAKE: {
             trace("Handling FUTEX_WAKE operation %i", operation);
             return _syscallhandler_futexWakeHelper(sys, uaddrptr, val);
         }
 
+        case FUTEX_WAIT_BITSET: {
+            trace("Handling FUTEX_WAIT_BITSET operation %i bitset %d", operation, val3);
+            if (val3 == FUTEX_BITSET_MATCH_ANY) {
+                return _syscallhandler_futexWaitHelper(
+                    sys, uaddrptr, val, timeoutptr, TIMEOUT_ABSOLUTE);
+            }
+            // Other bitsets not yet handled.
+            break;
+        }
+        case FUTEX_WAKE_BITSET: {
+            trace("Handling FUTEX_WAKE_BITSET operation %i bitset %d", operation, val3);
+            if (val3 == FUTEX_BITSET_MATCH_ANY) {
+                return _syscallhandler_futexWakeHelper(sys, uaddrptr, val);
+            }
+            // Other bitsets not yet handled.
+            break;
+        }
+
         case FUTEX_FD:
         case FUTEX_REQUEUE:
         case FUTEX_WAKE_OP:
-        case FUTEX_WAIT_BITSET:
-        case FUTEX_WAKE_BITSET:
         case FUTEX_LOCK_PI:
         case FUTEX_TRYLOCK_PI:
         case FUTEX_UNLOCK_PI:
         case FUTEX_CMP_REQUEUE_PI:
-        case FUTEX_WAIT_REQUEUE_PI:
-        default: {
-            warning("We do not yet handle futex operation %i", operation);
-            result = -ENOSYS; // Invalid operation specified in futex_op
-            break; // return SYSCALL_DONE
-        }
+        case FUTEX_WAIT_REQUEUE_PI: break;
     }
 
-    return (SysCallReturn){.state = SYSCALL_DONE, .retval.as_i64 = result};
+    warning("Unhandled futex operation %i", operation);
+    return (SysCallReturn){.state = SYSCALL_DONE, .retval.as_i64 = -ENOSYS};
```

Comment:
I think `int result = 0;` on line 158 is unused now and should be removed.

Reply:
Done